spec/defines/daemon_spec.rb

```diff
@@ -91,77 +91,32 @@
             it { is_expected.to contain_archive("/tmp/smurf_exporter-#{parameters[:version]}.tar.gz").with_extract_path('/opt/foo') }
           end
 
-          if ['centos-7-x86_64', 'centos-8-x86_64', 'debian-9-x86_64', 'debian-10-x86_64', 'redhat-7-x86_64', 'redhat-8-x86_64', 'ubuntu-16.04-x86_64', 'ubuntu-18.04-x86_64', 'archlinux-5-x86_64', 'virtuozzolinux-7-x86_64'].include?(os)
-            # init_style = 'systemd'
+          it { is_expected.to contain_class('systemd') }
 
-            it { is_expected.to contain_class('systemd') }
-
-            it {
-              is_expected.to contain_systemd__unit_file('smurf_exporter.service').that_notifies(
-                'Service[smurf_exporter]'
-              ).with_content(
-                %r{User=smurf_user\n}
-              ).with_content(
-                %r{ExecStart=/usr/local/bin/smurf_exporter\n\nExecReload=}
-              )
-            }
-
-            if Gem::Version.new(facts[:puppetversion]) < Gem::Version.new('6.1.0')
-              it { is_expected.to contain_class('systemd::systemctl::daemon_reload').that_comes_before('Service[smurf_exporter]') }
-            end
+          it {
+            is_expected.to contain_systemd__unit_file('smurf_exporter.service').that_notifies(
+              'Service[smurf_exporter]'
+            ).with_content(
+              %r{User=smurf_user\n}
+            ).with_content(
+              %r{ExecStart=/usr/local/bin/smurf_exporter\n\nExecReload=}
+            )
+          }
 
-            context 'with overidden bin_name' do
-              let(:params) do
-                parameters.merge(bin_name: 'notsmurf_exporter')
-              end
+          if Gem::Version.new(facts[:puppetversion]) < Gem::Version.new('6.1.0')
+            it { is_expected.to contain_class('systemd::systemctl::daemon_reload').that_comes_before('Service[smurf_exporter]') }
+          end
 
-              it {
-                is_expected.to contain_systemd__unit_file('smurf_exporter.service').with_content(
-                  %r{ExecStart=/usr/local/bin/notsmurf_exporter}
-                )
-              }
+          context 'with overidden bin_name' do
+            let(:params) do
+              parameters.merge(bin_name: 'notsmurf_exporter')
```

Comment:
Better to use `super()` here too
```suggestion
              super().merge(bin_name: 'notsmurf_exporter')
```